apis/object/v1alpha1/types.go

```diff
@@ -18,10 +18,61 @@ package v1alpha1
 
 import (
 	xpv1 "github.com/crossplane/crossplane-runtime/apis/common/v1"
+	"github.com/crossplane/crossplane-runtime/pkg/fieldpath"
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 	"k8s.io/apimachinery/pkg/runtime"
 )
 
+// A ManagementPolicy determines what should happen to the underlying external
+// resource when a managed resource is created, updated, deleted, or observed.
+// +kubebuilder:validation:Enum=Default;ObserveCreateUpdate;ObserveDelete;Observe
+type ManagementPolicy string
+
+const (
+	// Default means the provider can fully manage the resource.
+	Default ManagementPolicy = "Default"
+	// ObserveCreateUpdate means the provider can observe, create, or update
+	// the resource, but can not delete it.
+	ObserveCreateUpdate ManagementPolicy = "ObserveCreateUpdate"
+	// ObserveDelete means the provider can observe or delete the resource, but
+	// can not create and update it.
+	ObserveDelete ManagementPolicy = "ObserveDelete"
+	// Observe means the provider can only observe the resource.
+	Observe ManagementPolicy = "Observe"
+)
+
+// FromObject refers to an object by Name, Kind, APIVersion, etc. It is used
+// to reference other Object or arbitrary Kubernetes resource which is either
+// cluster or namespace scoped.
+type FromObject struct {
+	// APIVersion of the referenced object.
+	APIVersion string `json:"apiVersion"`
+	// Kind of the referenced object.
+	Kind string `json:"kind"`
+	// Name of the referenced object.
+	Name string `json:"name"`
+	// Namespace of the referenced object.
+	// +optional
+	Namespace string `json:"namespace,omitempty"`
+	// FieldPath is the path of the field on the resource whose value is to be
+	// used as input.
+	// +optional
+	FieldPath *string `json:"fieldPath,omitempty"`
```

Comment:
This just provides the flexibility as in some cases people might just need the dependency declaration to ensure the deployment sequence. Otherwise, people have to patch something although it might not really needed in such a case.

This is a bit similar to the idea of ["SyncWaves"](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-waves/) from ArgoCD where it uses annotation to declare the dependency (no patching), but ArgoCD can only control those resources that are stored in git repository. K8s provider doesn't have this limitation.

Reply:
I like the idea of using `dependsOn` and `patchesFrom`, code has been updated to reflect this.